src/ConsoleBuffer/Commands/ControlSequence.cs

```diff
@@ -36,14 +53,16 @@ public static Base Create(char command, string bufferData)
         }
 
         public bool IsExtended { get; private set; }
-        protected IList<string> Parameters { get; private set; }
+        protected ParsedParameters Parameters { get; private set; } = new ParsedParameters();
         protected ControlSequence(string bufferData) : base(bufferData) { }
 
         protected override void Parse(string bufferData)
         {
+            this.Parameters.Count = 0;
+
             if (bufferData.Length == 0)
             {
-                this.Parameters = Array.Empty<string>();
+                this.Parameters.Parameters[0] = 0;
```

Comment:
I don't think these are semantically identical, although in the case of VT sequence parsing they might be. A comment explaining this would be helpful.

Reply:
I've fixed up the 'reset' of my helper class.  Now I this.Parameters.Clear() before parsing begins and put's it into a good, known state.